source/common/config/cds_json.cc

```diff
@@ -181,7 +181,12 @@ void CdsJson::translateCluster(const Json::Object& json_cluster,
                                                 *cluster.mutable_tls_context());
   }
 
+  if (json_cluster.getString("features", "") == "use_downstream_protocol") {
+    cluster.set_protocol_selection(envoy::api::v2::Cluster::USE_DOWNSTREAM_PROTOCOL);
```

Comment:
Can we avoid putting this in v1? We're trying to restrict new features to v2 only to encourage migration, unless there is a direct need for someone to have it in v1.